cmake/CopyTarget.cmake

```diff
@@ -0,0 +1,145 @@
+# Provides the copy_target() function, which is used to make duplicate MPI and OpenMP versions of targets.
+
+# copies the value of a property from one target to another.  Does nothing if the property is not defined on the SOURCE.
+function(copy_property PROPERTY SOURCE DESTINATION)
+	get_property(PROP_IS_DEFINED TARGET ${SOURCE} PROPERTY ${PROPERTY} SET)
+	
+	if(PROP_IS_DEFINED)
+		get_property(PROP_VALUE TARGET ${SOURCE} PROPERTY ${PROPERTY})
+		#message("Copying ${PROPERTY} with value ${PROP_VALUE}")
+		set_property(TARGET ${DESTINATION} PROPERTY ${PROPERTY} "${PROP_VALUE}")
+	endif()
+	
+endfunction(copy_property)
+
+# Copies a target.  Creates a new target with a different name and the same sources and (important) properties.  Doesn't work on aliases or imported libraries.
+# Allows switching out sources if needed.
+# usage: copy_target(<target> <new name> [SWAP_SOURCES <source 1...> TO <replacement source 1...>])
+
+function(copy_target SOURCE DESTINATION)
+
+	#message("Copying target ${SOURCE} -> ${DESTINATION} --------------------------------------")
+	# parse arguments
+	# --------------------------------------------------------------------	
+	cmake_parse_arguments(COPY_TARGET "" "" "LANGUAGES;SWAP_SOURCES;TO" ${ARGN})
+	
```

Comment:
In the call `copy_target(libquick libquick_cuda SWAP_SOURCES TO $<TARGET_OBJECTS:cublas_fortran_obj> $<TARGET_OBJECTS:cusolver_obj> ${QUICK_CUDA_FORTRAN_SOURCE})`,
do `$<TARGET_OBJECTS:cublas_fortran_obj> $<TARGET_OBJECTS:cusolver_obj> ${QUICK_CUDA_FORTRAN_SOURCE}` all become ${COPY_TARGET_TO}, while ${COPY_TARGET_SWAP_SOURCES} and ${COPY_TARGET_LANGUAGES} are simply empty strings?

Reply:
Yep, you got it.